app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :name, presence: true, length: {minimum:6, maximum:25}
+  validates :email, presence: true, length: {maximum: 255}, uniqueness: true,
+            format: {with: VALID_EMAIL_REGEX}
+
+  before_save :downcase_email
+  private
+  def downcase_email
+    self.email = email.downcase
```

Comment:
```suggestion
    email.downcase!
```
downcase! khác gì với downcase e ?

Reply:
downcase! nó sẽ sửa đổi string tại chỗ

Reply:
còn gì nữa k e ?

Reply:
downcase! nó thay đổi chuỗi ban đầu, còn downcase thì không